ui/components-react/src/components-react/propertygrid/component/VirtualizedPropertyGridWithDataProvider.tsx

```diff
@@ -47,25 +47,41 @@ export interface VirtualizedPropertyGridWithDataProviderProps
 export function VirtualizedPropertyGridWithDataProvider(
   props: VirtualizedPropertyGridWithDataProviderProps
 ) {
-  const modelSource = usePropertyGridModelSource({
+  const [showSpinner, setShowSpinner] = useState(false);
+
+  const { modelSource, inProgress } = useTrackedPropertyGridModelSource({
     dataProvider: props.dataProvider,
   });
+
   const model = usePropertyGridModel({ modelSource });
   const eventHandler = usePropertyGridEventHandler({ modelSource });
+  useEffect(() => {
+    if (!inProgress) {
+      setShowSpinner(inProgress);
+      return;
+    }
+
+    // only set a timeout when inProgress is set to `true`
+    const timeout = setTimeout(() => {
+      setShowSpinner(inProgress);
+    }, 250);
 
-  if (!model) {
-    return (
-      <div className="components-virtualized-property-grid-loader">
-        <DelayedSpinner size="large" />
-      </div>
-    );
-  }
+    return () => clearTimeout(timeout);
+  }, [inProgress, model]);
 
   return (
-    <VirtualizedPropertyGrid
-      {...props}
-      model={model}
-      eventHandler={eventHandler}
-    />
+    <>
+      {showSpinner || !model ? (
```

Comment:
Why dont we simply use `inProgress` here instead of adding a new variable? `DelaySpinner` already have a builtin delay which could be adjusted with the `delay` prop if it is not of the right time.

Reply:
Because we wanted to have a delay on deciding when to switch between `VirtualizedPropertyGrid` and `DelaySpinner` to avoid flickering between the components. Instead of always switching to the spinner component when `inProgress===true`, we only switch when inProgress is true for more than a certain delay. This makes our component less _flickery_ if the requests are fast to complete.

Reply:
What is displayed by the Grid when the model is defined but the call are still `inProgress`, why is `model` even defined if we havent got the info we need, does it have any use ? It feels that this will cause much more flickering when the requests are not fast to complete. (but, maybe these are edgecases ?)

So, before this PR:
 - If there is a `model`, show the `VPGrid`. (what is actually displayed here if not ready ?)
 - If there is no `model`, show the `DelayedSpinner`, so an empty div.
   - After `500ms`, show `Spinner` in that div.

Now:
 - If there is no `model`, show the `DelayedSpinner`, so an empty div.
   - After `500ms`, show `Spinner` in that div.
 - If there is a `model`, show the `VPGrid`.
   - If `inProgress`, after `150ms`, remove `VPGrid` (what was the grid displaying for those 150ms ?) show the `DelayedSpinner`, so an empty div.
     - After `500ms`, show `Spinner` in that div.

Reply:
The implementation has been updated to use a custom renderer for a spinner so that we wouldn't have timeouts in different places. 

The workflow after this PR:
- If there is no model, show `ProgressRadial` immediately, no empty div. (Check for model being defined is only relevant for the first `getData` call as it is defined afterwards)
- If there is a defined model and `inProgress` is false, show the `VPGrid`
  - If `inProgress`, after `250ms` render 'ProgressRadial', so no empty div.

Hope this clears things up, let me know if I missed something.

Reply:
Yes, sorry I missed the change from `DelayedSpinner` to `ProgressRadial` within the `DelayedLoaderRenderer` component. 

> Could you expand on why you think so? (If you still think that is the case)

What confuses me is that while we are waiting for the loader to display, we display the grid, which, according to your answer above, may have content already as `model` may be defined. So the user will see data as we have a `model` but are still in progress, for a split second while `inProgress` is `true` at the same time that `model` is defined, then loose everything and that will be replaced by a spinner until `inProgress` turns `false`.

Reply:
Thanks for clarifying.

If I understood correctly you are talking about situations where the window between delay ending and the `getData` requests completing is rather small. Lets say delay of waiting before showing the loader is `250ms` and requests are completing within `251 to 350ms`, then yes, the component will look a bit flickery from switching between loader and `VPGrid` rapidly. 

However, I believe these are edge cases and I don't think there is a simple way to remedy it without removing this feature altogether.